### ISSUE TITLE
Support inheritance for child states for ui-router 

### ISSUE THREAD

**Issue by mplushnikov** [vinaygopinath/ngMeta#14]
Hello, very nice service!
What do you think about support for inheritance for child states in ui-router?
At the moment it doesn't work for 'title' for example, but does for 'description'. I think this is why you are resetting the current state of ngMeta object in readRouteMeta() function.
It would be really nice enhancement!

**Comment by vinaygopinath** [comment]
Hi, thanks for creating an issue, and sorry for the delay in replying to this feature request - I like the idea of inheritance for ui-router child states, but work has kept me busy lately, so I probably won't be able to work on it in the next month + I'm not sure how to go about it (Is there any documentation on how a ui-router child state can access its parent's properties?). A PR would be appreciated :stuck_out_tongue:  

@ "Resetting the current state of ngMeta object": I'm not sure what you're referring to. 
`readRouteMeta` 
- uses a state's meta object, or falls back to an empty object.
- sets the title and titleSuffix based on the state's meta object (if one exists)
- deletes the title and titleSuffix properties in the meta and default objects
- sets the values for other meta properties defined for the state
- sets the default values for other default properties defined during configuration

**Comment by martyzz1** [comment]
I think this is easy to implement.
The crux of the issue is here:-

https://github.com/angular-ui/ui-router/wiki/Nested-States-%26-Nested-Views#inherited-custom-data

Instead of using 

```
.config(function ($routeProvider, ngMetaProvider) {

  $routeProvider
  .when('/home', {
    templateUrl: 'home-template.html',
    meta: {
      'title': 'Home page',
      'description': 'This is the description shown in Google search results'
    }
  })
  .when('/login', {
    templateUrl: 'login-template.html',
    meta: {
      'title': 'Login page',
      'titleSuffix': ' | Login to YourSiteName',
      'description': 'Login to the site'
    }
  });
});
```

You should Specify "meta" underneath the 'data' property. Then it will be inherited automatically.

```
.config(function ($routeProvider, ngMetaProvider) {

  $routeProvider
  .when('/home', {
    templateUrl: 'home-template.html',
    data: {
      meta: {
        'title': 'Home page',
        'description': 'This is the description shown in Google search results'
      }
    }
  })
  .when('/login', {
    templateUrl: 'login-template.html',
    data: {
      meta: {
        'title': 'Login page',
        'titleSuffix': ' | Login to YourSiteName',
        'description': 'Login to the site'
      }
    }
  });
```

As I'm migrating our codebase from a mechanism that already had this setup, I'm going to write a version that should support this and I'll raise you a PR. It should be possible to check $state.current.meta before $state.current.data.meta to keep things backwards compatible.

**Comment by mplushnikov** [comment]
Great! Thank you for the update!
Please only check README.md, the documentation for this great feature is not correctly visible at the moment!

**Comment by vinaygopinath** [comment]
Hey @mplushnikov, thanks for letting me know. I've updated the README and published a new version (v1.0.0) on npm with inheritance support.
